### PR TITLE
Scaffold Redux setup for Chat

### PR DESCRIPTION
**Pull request by bomanimc**
One challenge from the Dial Up Chat was the messy business we got into by trying to pass state values and functions to update state around our component hierarchy. To simplify this, we can use a tool like Redux that will help define a cleaner pattern for managing state across all of our components. 

This PR adds a basic Redux setup for chat that works with the Next.js. It also includes a simple example on the now-showing page that adds a new value to a list in Redux when the button is clicked. Other users who are connected to the socket will also receive your update.

If we decide to merge this, I'd highly recommend Redux DevTools browser extension (https://github.com/reduxjs/redux-devtools)! It makes it much easier to see what's happening in the redux actions and state.

### CODE REVIEW

**Comment by jbirori** [comment]
I actually haven't used Redux before so this is all @sehmon  or @MagicWilliams

**Comment by MagicWilliams** [comment]
Yo! Just took a look and things seem to be working well for me 👍

**Comment by MagicWilliams** [comment]
@bomanimc i am curious though, which part of this PR enables the app to update when someone else has sent a message to the chat? I see that functionality working, but not sure where that happens in the code & wanna check it out

**Comment by MagicWilliams** [comment]
was thinking it might be that useEffect hook you have?

File: state/actions.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as actionTypes from './actionTypes';
+
+// eslint-disable-next-line import/prefer-default-export
+export const sendMessage = (socketInstance, message) => (dispatch) => {
+  socketInstance.emit('message', message);
```
Comment (bomanimc) [inline state/actions.js:5]: @MagicWilliams this line essentially takes an instance of a socket connection and sends the message that the person typed in to the server using a websocket!

File: pages/now-showing.js [hunk 2]
```diff
@@ -10,13 +12,20 @@ const socketURL =
 const socket = io(socketURL);
 
 export default function NowShowing() {
+  const dispatch = useDispatch();
+  const messages = useSelector((state) => state.chat.messages);
+
   useEffect(() => {
```
Comment (bomanimc) [inline pages/now-showing.js:18]: We actually don't really need this useEffect hook and it's probably confusing to leave it here. Will remove.

**Comment by bomanimc** [comment]
Thinking about this more, it'd probably make sense to create a separate class that wraps our Socket.io stuff on the client-side and provides some methods that make it more straightforward to send different types of messages around. We probably will also want a Message class too. I can add these in a PR later today unless someone gets to it first!

**Pull request merged**